components/infra-proxy-service/server/nodes.go

```diff
@@ -327,9 +329,13 @@ func (s *Server) GetNodeExpandedRunList(ctx context.Context, req *request.NodeEx
 
 	runList := res.RunList
 
-	// Fetches cookbooks to evaluate recipes version.
+	if res.PolicyGroup != "" {
+		req.Environment = "_default"
+	}
+
 	cookbooks, err := c.client.Environments.ListCookbooks(req.Environment, "1")
 	if err != nil {
+		logrus.Errorf("Failed while listing cookbooks for default: %+v", err)
```

Comment:
you should not hard code the environment as default here, it should be from req.Environment